### ISSUE TITLE
Reading rlib on macOS

### ISSUE THREAD

**Issue by RazrFalcon** [mdsteele/rust-ar#6]
I'm using `ar` to read rust libraries (`.rlib`). Under Linux everything is fine, but I got an error under macOS:
```
Error { repr: Custom(Custom { kind: InvalidData, error: StringError("Invalid file size field in entry header (\"    644   \")") }) }
```

Is it possible to read `rlib`?

**Comment by mdsteele** [comment]
Hmm, that's odd.  I just tried it on a couple `.rlib` files generated on my OS X machine, and it worked fine.  Can you attach an example of an `.rlib` file that doesn't work?

**Comment by RazrFalcon** [comment]
I got this error on any `rlib` from macOS.

[libatty.zip](https://github.com/mdsteele/rust-ar/files/1650092/libatty.zip)

**Comment by mdsteele** [comment]
Thanks for the file.  Strangely, running `cargo run --example extract -- libatty-d1be7f28c1fcb94e.rlib` still works fine for me, though.  Does that command give an error for you?

What version of the `ar` crate are you using?  Also, what version of `rustc`?

**Comment by RazrFalcon** [comment]
I'm using 0.5 tag archive and run the rlib trough the `symbols` example.

**Comment by mdsteele** [comment]
Ah, thanks!  I'm able to reproduce the issue now.

I'll take a look and try to figure out what's going wrong.

**Comment by mdsteele** [comment]
Okay, the error should be fixed by commit e593551 -- I'm now able to run that `.rlib` through the `symbols` example without getting an error, and also now there are some new regression tests for that bug.

Separately, however, there's the issue that the library doesn't yet support reading symbol tables from BSD-variant archives (which is the format variant typically used by Mac, as opposed to the GNU variant typically used by Linux), so although the `symbols` example shouldn't crash anymore, it'll just give you an empty list of symbols, which isn't very helpful.

I'll see about getting BSD-variant symbol table support added.

**Comment by mdsteele** [comment]
Just pushed commit d43c201, which adds initial support for reading BSD-variant symbol tables.  The `symbols` example should hopefully now give useful output for `.rlib` files on Mac.
